pkg/util/prioritymap/map.go

```diff
@@ -37,13 +39,34 @@ func WithSeparator(separator string) Option {
 	}
 }
 
-// WithAllowDuplicatesForKey allows to add multiple values for the same key (covers all keys)
 func WithAllowDuplicates() Option {
 	return func(a *entry) {
 		a.allowDuplicates = true
 	}
 }
 
+func WithAvoidDuplicates() Option {
+	return func(a *entry) {
+		a.allowDuplicates = false
+	}
+}
+
+func WithAvoidDuplicatesFor(key string) Option {
+	return func(a *entry) {
+		if a.key == key {
+			a.allowDuplicates = false
+		}
+	}
+}
```

Comment:
nitpick:
I guess this was added for the sake of completeness, but I don't see an actual scenario where we want to avoid duplicates for only a few entries 😅

Reply:
It was actually part of the first approach to fix it, but I see no harm in keeping it. Makes the map more versatile.